lego/apps/polls/tests/test_api_polls.py

```diff
@@ -0,0 +1,98 @@
+        self.client.force_authenticate(self.unauthenticated_user)
+        response = self.client.post(_get_list_url(), self.poll_data)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_list_poll_authenticated(self):
+        """A user with permissions should be able to list all polls"""
+        self.client.force_authenticate(self.authenticated_user)
+        response = self.client.get(_get_list_url())
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_list_poll_unathenticated(self):
+        """A user with no permissions should be able to list all polls"""
+        self.client.force_authenticate(self.unauthenticated_user)
+        response = self.client.get(_get_list_url())
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_detailed_poll_authenticated(self):
+        """A user with permissions should not be able to retrive the detailed poll view"""
+        self.client.force_authenticate(self.authenticated_user)
+        response = self.client.get(_get_detail_url(1))
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_detailed_poll_unauthenticated(self):
+        """A user with no permissions should be able to retrive the detailed poll view"""
+        self.client.force_authenticate(self.unauthenticated_user)
+        response = self.client.get(_get_detail_url(2))
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_vote_poll_authenticated_unanswered(self):
+        """A user with permissions should be able to vote on a poll that he has not answered"""
+        self.client.force_authenticate(self.authenticated_user)
+        response = self.client.post(_get_vote_url(2), {"optionId": 3})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_vote_poll_authenticated_answered(self):
+        """A user should not be able to vote on a poll more than once"""
+        self.client.force_authenticate(self.authenticated_user)
+        self.client.post(_get_vote_url(2), {"optionId": 3})
+        response = self.client.post(_get_vote_url(2), {"optionId": 4})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+    def test_vote_poll_unauthenticated(self):
+        """A user with no permissions should be able to vote on a poll"""
+        self.client.force_authenticate(self.unauthenticated_user)
+        response = self.client.post(_get_vote_url(1), {"optionId": 1})
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_vote_poll_not_valid(self):
+        """A user should be able to vote on a poll where valid_until is not in the future"""
+        self.client.force_authenticate(self.authenticated_user)
+        response = self.client.post(_get_vote_url(3), {"optionId": 5})
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
Maybe add a test for `test_edit_poll_without_permissions` (and create), with normal useres without permissions to edit/create polls?